src/content/developers/docs/smart-contracts/formal-verification/index.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Formal verification of smart contracts
+description: An overview of formal verification techniques for Ethereum smart contracts
+lang: en
+sidebar: true
+---
+
+[Smart contracts](/developers/docs/smart-contracts/) are making it possible to create decentralized, trustless, and robust applications that introduce new use-cases and unlock value for users. Because smart contracts handle large amounts of value, security is a critical consideration for developers.
+
+Reliability is a highly desired quality of smart contracts, and both users and developers want assurances that a smart contract will work as intended. This is important because code in the [Ethereum Virtual Machine (EVM)](/developers/docs/evm/) is immutable, which rules out the possibility of fixing bugs or retrieving funds lost to malicious attacks.
+
+Formal verification is one of the recommended techniques for [testing smart contracts](/developers/docs/smart-contracts/testing/) and improving [smart contract security](/developers/docs/smart-contracts/security/). Formal verification involves testing a program (a smart contract in this case) to ensure it behaves as expected during execution.
+
+## What is formal verification? {#what-is-formal-verification}
+
+Formal verification is used to evaluate the _correctness_ of software programs and confirm compliance with predefined specifications using formal methods of mathematics. [More on formal verification](https://en.wikipedia.org/wiki/Formal_verification).
+
+In the context of Ethereum, formal verification can prove that a smart contract will only implement the desired business logic.
+
+### Why formal verification is necessary {#why-formal-verification-is-necessary}
+
+While manual code audits are great, human auditors cannot possibly cover every possible vulnerability. Also, an audit may reveal bugs in contract code, but it cannot confirm the absence of vulnerabilities.
+
+Formal verification is grounded in [formal methods](https://en.wikipedia.org/wiki/Formal_methods), which use mathematically rigorous methods for specifying, developing, and verifying software. Formal verification allows developers to mathematically model software programs and create formalized specifications, making testing more precise and robust.
+
+Specifications are properties or used to describe the intended behavior of a smart contract. Formal verification can prove that a contract behaves exactly as described in the specification. Unlike testing based on informal assumptions, formal verification provides mathematical proof of a smart contract's correctness.
```

Comment:
> Specifications are properties or used to

This sentence doesn't make sense to me. Should it be "Specifications are properties used to", or something else?

Reply:
It's a typo. The sentence should read "specifications are properties used to...". I'll edit the content to fix it.